src/topology.jl

```diff
@@ -190,3 +190,121 @@ function Cart_sub(comm::Comm, remain_dims)
     end
     comm_sub
 end
+
+######### vvv these functions are are not recommended anymore! vvv #########
+# function Graph_create(comm::Comm, indices::Vector{Cint}, edges::Vector{Cint}, reorder::Cint=0)
+#     graph_comm::Comm
+# # int MPI_Graph_create(MPI_Comm comm_old, int nnodes, const int indx[],
+# #                      const int edges[], int reorder, MPI_Comm * comm_graph)
+#     @mpichk ccall((:MPI_Graph_create, libmpi), Cint,
+#          (MPI_Comm, Cint, Ptr{Cint}, Ptr{Cint}, Cint, Ptr{MPI_Comm}),
+#          comm,length(indices),indices,edges,reorder,graph_comm)
+#     return graph_comm
+# end
+
+# function Graph_neighbors_count(graph_comm::Comm, rank::Cint)
+#     neighbors_count::Cint
+# # int MPI_Graph_neighbors_count(MPI_Comm communicator, int rank,
+# #                              int* neighbours_count)
+#     @mpichk ccall((:MPI_Graph_neighbors_count, libmpi), Cint,
+#         (MPI_Comm, Cint, Ptr{Cint}),
+#         comm,rank,neighbors_count)
+#     return neighbors_count
+# end
+
+# function Graph_neighbors(graph_comm::Comm, rank::Cint, neighbors::Array{Cint})
+# # int MPI_Graph_neighbors(MPI_Comm communicator, int rank,
+# #                         int neighbours_count, int neighbours[])
+#     @mpichk ccall((:MPI_Graph_neighbors_count, libmpi), Cint,
+#         (MPI_Comm, Cint, Cint, Ptr{Cint}),
+#         comm,rank,neighbors,neighbors)
+#     return neighbors
+# end
+######### ^^^ these functions are are not recommended anymore! ^^^ #########
+
+function Dist_graph_create_adjacent(comm::Comm, sources::Vector{Cint}, source_weights::Vector{Cint}, destinations::Vector{Cint}, destination_weights::Vector{Cint}, info::Info, reorder=0)
```

Comment:
We have a convention for using keyword arguments for `Info` objects (e.g. see the IO functions)

Reply:
also `reorder` can simply be a `Bool` (`ccall` handles the conversion)

Reply:
Thanks for the information. Learned again something new. :)